### DISCUSSION TITLE
oznoz.com - Downloads Video without any audio?

### DISCUSSION THREAD

**Discussion by mcom-gb** [aclap-dev/video-downloadhelper#279] [Help and Q&A]
### Your question

Hello,
I have the latest versions - registered on Brave and Firefox - however on oznoz.com I am getting a download of the video but without any audio stream in the file at all?

I've tried using Firefox, Brave (Chrome I dont think supports it) and each and every file downloaded 
(eg from this directory, registration is free) downloads the video without the audio stream.
https://video.oznoz.com/video/rita-and-crocodile-season-1.html

It would be really appreciated - this is a video series that I watched with my late wife and my son, and I've tried to purchase from every source possible...

many thanks,
M

### Addon & CoApp version and details

version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): e7c90c884ada4359
platform: aarch64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "arm64",
    "node": "18"
  },
  "home": "/Users/Cuellar",
  "converterBinary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "n6.1-11-gc30f360955"
}

**Comment by Mattmjs** [comment]
Share the hit details information, it should have a master manifest & a media manifest.

**Comment by mcom-gb** [comment]
bitrate | 2073520
-- | --
chunked | hls
core_media | {...}
created | 1710241058883
descrPrefix | HLS streaming
duration | 311
extension | mp4
group | grp-14f1bd8f
headers | [...]
id | hls:7e05c4f5
isPrivate | false
masterManifest | https://d3mm5apfntsy6t.cloudfront.net/Dansk/RAC/S01_01/English/001/RAC_S01E01_ENG_ios_Main.m3u8
mediaManifest | https://d3mm5apfntsy6t.cloudfront.net/Dansk/RAC/S01_01/English/001/RAC_S01E01_ENG_ios_Main/1.87Mbps.m3u8
originalId | network-probe:51eb6fa0
pageUrl | https://video.oznoz.com/video/rita-and-crocodile-season-1.html
referrer | https://video.oznoz.com/video/rita-and-crocodile-season-1.html
size | 1024x768
status | active
tabId | 1439985367
thumbnailUrl | https://video.oznoz.com/media/brands/1643238849_RitaandCrocodile_Banner_768x400.png
title | Rita and Crocodile
topUrl | https://video.oznoz.com/video/rita-and-crocodile-season-1.html

thankyou!!

**Reply by Mattmjs** [reply]
Ok so doesn't look like it's getting the audio with the video like you say. ffprobe says it couldn't find codec parameters for the audio.

I'll attach the manifest and ffprobe information. However I used another tool **yt-dlp** which handled the download with audio.
Why it worked I don't know. 

[RAC_S01E01_ENG_ios_Main.txt](https://github.com/aclap-dev/video-downloadhelper/files/14571776/RAC_S01E01_ENG_ios_Main.txt)
[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/14571793/ffprobe.txt)

**Reply by Mattmjs** [reply]
@paulrouget , you might be able to see what's happening on this one.
@matt-gb , if you've used yt-dlp before you'll be able to get what your after. If not I'll show you tomorrow.

**Reply by GrampaWildWilly** [reply]
This is rather odd.  When I went to the oznoz link in the original post & clicked on episode 1, I got a login prompt.  In other words, I can't test VDH on this video.  But, the HLS master manifest in @matt-gb's latest post above is completely accessible to me.  I ffprobed it:

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/14583395/ffprobe.txt)

Then I used that information to download the video with ffmpeg:

[Danish Video mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14583399/Danish.Video.mp4.Log.txt)

It played fine in VLC, both video & audio, although I only sampled it, didn't sit & watch it.  It's kind of weird that I can't get into the site without login credentials but if you supply an HLS master manifest, I can get the video with ffmpeg.  Seems like faulty site security.  Whatever.

I think this is encouraging that VDH ought to be able to get this content.  But the requirement to log in might be an impediment for @paulrouget.

@Mattmjs The links in your post are giving me 404 Not Found errors.  Hmmm.....  I'm getting the same error on my links in this post.  I think maybe GitHub is having some site problems.  Weird . . .

**Reply by Mattmjs** [reply]
Did you not get an any errors on the audio in ffprobe like this:

[hls @ 000002979e1c3600] Could not find codec parameters for stream 1 (Audio: aac ([15][0][0][0] / 0x000F), 0 channels, fltp): unspecified sample rate
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options

That's what I had attached. It's a strange one, it doesn't work in **Version 8.2.1.1 (stable)**. Video is downloaded, audio missing.
But in **Version 8.2.2.7a1 (beta)** I get a video with audio. 

So I think Paul doesn't need to look into this now, @matt-gb  just needs the beta extension and it'll work fine.

**Reply by GrampaWildWilly** [reply]
The problem with attachments was temporary.  I can see your attachments now & mine as well.

No, I didn't get the errors on the audio tracks because of a hint @paulrouget gave in another thread a week or so ago.  Now that you can see my ffmpeg log, look for my parameters named:

analyzeduration
probesize

The error messages you got actually tell you what the solution is.  I had no idea this was actually relevant until Paul pointed it out to me.  The first time I used those 2 parameters, an apparently faulty audio stream suddenly was no problem at all.  Now I routinely include those 2 parameters on every ffprobe & ffmpeg invocation I issue.  I think Paul added those parameters to the CoApp's ffmpeg . . . or maybe he modified the way the CoApp invokes ffmpeg by adding those parameters.  I don't know what values he's chosen.  The numbers you can see in my log are kind of made up.  I tried a range of numbers before I got to something that worked.  I then decided to just make them ridiculously large so I could forget about them.

How are you getting into the oznoz site & using VDH if you don't have a membership there?

**Reply by Mattmjs** [reply]
I'll have to look at that log, I've hardly ever come across errors like that with the audio. 
How did I get into the site ? I didn't use the site.
It's a little trick where I paste the manifest directly in Firefox, the VDH button becomes colored indicating it detected the
streams from that manifest. That's why I ask for the hit details. Getting a manifest allows to bypass logging into a site.

**Reply by GrampaWildWilly** [reply]
These manifests for oznoz don't have any of the gobbledygook strings that so many of these URLs do.  I wonder if the trick works on one of these that has the string.  In such a case, I imagine you'd be on a clock & if you don't use whatever the person posted within a certain amount of time, the manifest would quit working because it timed out.

**Reply by mcom-gb** [reply]
The login is purely email registration, there is no need on the content I looked at to pay at all..
...Could I just get hold of the **Version 8.2.2.7a1 (beta)** (Chrome and Firefox plse - I have both registered)

As the OSX version of yt-dlp I struggled to install via bash last night...despite multiple attempts

many many thanks!
M

**Reply by Mattmjs** [reply]
Here is where the beta versions are, I'm not sure if the chrome beta works the same as the Firefox beta.

https://github.com/aclap-dev/video-downloadhelper/wiki/Beta-versions/

**Comment by mcom-gb** [comment]
Brilliant - The Beta works (at least on Firefox!) 

SOOO much appreciated..!

**Comment by GrampaWildWilly** [comment]
It took me several search attempts but I finally found the discussion where Paul told me about those parameters for recognizing apparently corrupt audio that is in fact not corrupt: #238.
